src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1304,6 +1305,44 @@ def create_or_replace_view(
             source_plan,
         )
 
+    def find_table_function_in_sql_tree(self, plan: SnowflakePlan):
+        from snowflake.snowpark._internal.analyzer.select_statement import (
+            SelectTableFunction,
+            Selectable,
+        )
+
+        if isinstance(plan, SelectTableFunction) and isinstance(
+            plan.snowflake_plan.source_plan, TableFunctionJoin
+        ):
+            if (
+                plan.snowflake_plan.source_plan.right_cols == ["*"]
+                and len(plan.snowflake_plan.children_plan_nodes) == 1
+            ):
+                child_plan = plan.snowflake_plan.children_plan_nodes[0]
```

Comment:
please add comments here to explain what we are trying to extract from `TableFunctionJoin` node